### DISCUSSION TITLE
Next Learn Create Nextjs Blog App

### DISCUSSION THREAD

**Discussion by reveurguy** [vercel/next.js#38751] [Help] [answered]
I am following the learn course to create a nextjs blog app. 

This is the index page of the sample app.
<img width="688" alt="Screenshot 2022-07-18 at 1 44 01 PM" src="https://user-images.githubusercontent.com/59417546/179470550-fbabdc42-4d49-4427-9359-4f5f3d239f94.png">

This the blog page
<img width="979" alt="Screenshot 2022-07-18 at 1 44 26 PM" src="https://user-images.githubusercontent.com/59417546/179470627-1fb58f2a-e95a-4fb2-9cb3-f6293e156586.png">

It's code can be found here : https://github.com/vercel/next-learn/tree/master/basics/demo

In the blog page, I want to display images related to the blog at the top instead of name and profile image. I have created a layout for that but am having a problem on how i can set the src in Image (next/image) in the layout to use images particularly for that blog.

**Comment by valcosmos** [comment, marked as answer]
Hello,  I ran this example, you can replace this place in the picture with the image you want related to your blog.

![image](https://user-images.githubusercontent.com/53572196/179475158-69ffd0c7-021c-4c25-8085-e3afb2567d9b.png)

**Reply by reveurguy** [reply]
No actually, i got that. What I am asking is that how can I set the src to search for image name which is same as blog name and display that image. If i hard code any src here, that particular will display on all blogs, I don't wanna do that. I want to add an image for every blog i add and display the particular image for a blog on that blog page.

**Reply by valcosmos** [reply]
I'm sorry, I didn't understand what you meant just now.

Regarding your problem, you can try this:

In `pages/posts/[id].js`

```ts
export default function Post({ postData }) {
  return (
//======set the URL of the image you want to use, If postData has image attribute=======
    <Layout image={ postData.image }>
//==============
      <Head>
        <title>{postData.title}</title>
      </Head>
      <article>
        <h1 className={utilStyles.headingXl}>{postData.title}</h1>
        <div className={utilStyles.lightText}>
          <Date dateString={postData.date} />
        </div>
        <div dangerouslySetInnerHTML={{ __html: postData.contentHtml }} />
      </article>
    </Layout>
  )
}
```

In `components/layout.js`

```js
export default function Layout({ children, home, title, image }) {
  return (
    <div className={styles.container}>
      <header className={styles.header}>
        {home ? (
          <>
            <Image
              priority
              src="/images/profile.jpg"
              className={utilStyles.borderCircle}
              height={144}
              width={144}
              alt={name}
              unoptimized={true}
            />
            <h1 className={utilStyles.heading2Xl}>{name}</h1>
          </>
        ) : (
          <>
            <Link href="/">
              <a>
                <Image
                  priority
//===========use the image here==========
                  src={image}
//=====================
                  className={utilStyles.borderCircle}
                  height={108}
                  width={108}
                  alt={name}
                />
              </a>
            </Link>
            <h2 className={utilStyles.headingLg}>
              <Link href="/">
                <a className={utilStyles.colorInherit}>{title}</a>
              </Link>
            </h2>
          </>
        )}
      </header>
    </div>
  )
}
```

If you have an image in each article data, you can use it like this, and then the top of each article will be the corresponding article image.

**Comment by reveurguy** [comment]
yeah thanks. that worked.
just needed to add banner attribute in md file with the image path as it's value and move the image to the same folder as the md file.

**Comment by reveurguy** [comment]
Also wanted to ask one small thing, i added a table of contents to my blog, and as the .md file was parsed and converted to html, it doesn't work when i deploy to the site. Any way i can make it work? On the site when i click on any heading in the table of contents, it wants to go to #heading but since it is `<h1>heading</h1> `on the site, it just goes to top of page like when we add src="#" to a button.

**Reply by valcosmos** [reply]
Hello, Is there an error in the console?

**Reply by reveurguy** [reply]
No there is no error.

**Reply by valcosmos** [reply]
Hello, If the console has no errors, I don't have enough information to deal with this, i am so sorry.

**Comment by reveurguy** [comment]
how can i use a js script?
i tried adding it to head in the layout js file but that didn't work, i tried adding it directly in [id].js file in the return part of Post function but that didn't work.
the script is adding id to the headings on the page.

**Reply by valcosmos** [reply]
Hello, JS scripts should be written in the `_document.js` file.

```js

// pages/_document.js
import { Html, Head, Main, NextScript } from 'next/document'

export default function Document() {
  return (
    <Html>
      <Head>
      // write here
      </Head>
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  )
}
```

**Comment by reveurguy** [comment]
i have a script js file.
i am trying to add the script tag with src= location of script js file in the above document js file
but the console is showing this error: net::ERR_ABORTED 500 (Internal Server Error)
where should i place that file?

**Comment by reveurguy** [comment]
I was able to achieve what i wanted to by using inline script from next/script but the script is not running as it should

**Comment by reveurguy** [comment]
this is my script
```
var h1Heading= document.getElementsByTagName("h1");
            for(let i=1; i<h1Heading.length; i++) {
              let str = h1Heading[i].innerText;
              str = str.replace(/[^a-zA-Z ]/g, "");
              str = str.replace(/\s+/g, '-').toLowerCase();
              h1Heading[i].id = str;
            }
```
it should add an id to to h1 according to it's text.
If h1 text is: What is ABC?
it should set the id="what-is-abc"
but the id value is not setting to that

**Reply by valcosmos** [reply]
Hello, u can try setting this property in `Script` tag.

<img width="735" alt="image" src="https://user-images.githubusercontent.com/53572196/179747697-665aa1fa-1c4a-48aa-800c-1670a92c8f90.png">
